### DISCUSSION TITLE
Espressif-IDE v3.0.0 — LSP Support for C/C++ Editor · Espressif Developer Portal

### DISCUSSION THREAD

**Discussion by giscus** [espressif/developer-portal#210] [General]
# Espressif-IDE v3.0.0 — LSP Support for C/C++ Editor · Espressif Developer Portal

The developer resources in just one place!

https://developer.espressif.com/blog/espressif-ide-lsp-support/

**Comment by klausbng** [comment]
Definitely added too early. The editor seems to be an alpha release never tested by someone who wants to use it for developing C-code. Never cursed so much when programming: I have configured the CDT editor to my needs graphically. The new one seems to need awkward editing a configuration file, which I could not find. It sets semicolons, when I need brackets, It sets line starts at positions, which are wrong, it uses tabs, when I would like to set spaces. If I open a second view to see the area of my code containing definitions, trying to edit in the second view, the first view permanently jumps back to the position of the cursor. 
There are some people out there, who have to earn money with developing and don't want to waste time with dealing with half-baked software.

**Reply by emaayan** [reply]
are you talking about formatting? cause that's clang-format and I've opened several issues about it.

**Comment by WalkaboutJw** [comment]
I would rather liked to see fixes for some indexer issues then introducing something so unstable as the clang stuff is.
It creats lots of false errors and even when it works is the implausible code completition more a hindering then a help.
